src/attributes/utils/handlers.ts

```diff
@@ -182,39 +182,49 @@ export const prepareAttributesInput = ({
   updatedFileAttributes
 }: AttributesArgs): AttributeValueInput[] =>
   attributes.map(attribute => {
-    if (attribute.data.inputType === AttributeInputTypeEnum.FILE) {
-      const updatedFileAttribute = updatedFileAttributes.find(
-        attributeWithNewFile => attribute.id === attributeWithNewFile.id
-      );
-      if (updatedFileAttribute) {
+    switch (attribute.data.inputType) {
+      case AttributeInputTypeEnum.FILE:
+        const updatedFileAttribute = updatedFileAttributes.find(
+          attributeWithNewFile => attribute.id === attributeWithNewFile.id
+        );
+
+        if (updatedFileAttribute) {
+          return {
+            file: updatedFileAttribute.file,
+            id: updatedFileAttribute.id
+          };
+        }
         return {
-          file: updatedFileAttribute.file,
-          id: updatedFileAttribute.id
+          file:
+            attribute.data.selectedValues &&
+            attribute.data.selectedValues[0]?.file?.url,
+          id: attribute.id
+        };
```

Comment:
It's kinda too big for switch - let's extract it to separate function and return its result here